### PR TITLE
Add warning messages when trying to assert Selector/ClientFunction instances

### PR DESCRIPTION
**Pull request by Ogurecher**
## Purpose
Users often want to assert that the selected element exists or that the returned value of ClientFunction meets a specific criteria.

## Approach
I created an addAssertionWarning function that checks if the "actual" expected value is a function and if its builder object is SelectorBuilder/ClientFunctionBuilder. If this condition is met, I print a corresponding warning.

## References
Part of #5087

## Pre-Merge TODO
- [x] Write tests for your proposed changes
- [x] Make sure that existing tests do not fail

### CODE REVIEW

File: src/api/test-controller/index.js [hunk 1]
```diff
@@ -300,6 +306,19 @@ export default class TestController {
     _expect$ (actual) {
         const callsite = getCallsiteForMethod('expect');
 
+        const addAssertionWarning = (value, builderClassName, warningMessage, warningCallsite) => {
```
Comment (AndreyBelym) [inline src/api/test-controller/index.js:309]: I think we can simplify this code in the following way:
```js
_expect$ (actual) {
   //....
    if (isClientFunction(actual))
        this._addWarning(WARNING_MESSAGE.assertedClientFunctionInstance, callsite);

    if (isSelector(actual))
        this._addWarning(WARNING_MESSAGE.assertedSelectorInstance, callsite);
}

_addWarning (message, callsite = void 0) {
    const renderedCallsite = renderCallsiteSync(...);
    
    this.warningLog.addWarning(...)
}
```
`isSelector` and `isClientFunction` can be placed in an external module like `src/client-functions/types.ts`. You can declare them as type guards.

File: src/api/test-controller/index.js [hunk 2]
```diff
@@ -300,6 +306,19 @@ export default class TestController {
...
+            const renderedCallsite = renderCallsiteSync(warningCallsite, {
+                renderer:    renderers.noColor,
+                stackFilter: createStackFilter(Error.stackTraceLimit)
+            });
+
+            if (typeof value === 'function' && value[Object.getOwnPropertySymbols(value)[0]] && value[Object.getOwnPropertySymbols(value)[0]].constructor.name === builderClassName)
```
Comment (AndreyBelym) [inline src/api/test-controller/index.js:315]: You can import `functionBuilderSymbol` directly from 'src/client-functions/builder-symbol.js`

File: src/api/test-controller/index.js [hunk 3]
```diff
@@ -345,9 +351,24 @@ export default class TestController {
     _expect$ (actual) {
         const callsite = getCallsiteForMethod('expect');
 
+        if (isClientFunction(actual))
+            this._addWarning(WARNING_MESSAGE.assertedClientFunctionInstance, callsite);
+
+        if (isSelector(actual))
```
Comment (miherlosev) [inline src/api/test-controller/index.js:357]: `else if (isSelector(actual))`

File: src/notifications/warning-message.js [hunk 4]
```diff
@@ -32,6 +32,8 @@ export default {
                                                         'The placeholder{suffix} {verb} replaced with an empty string.',
 
     clientScriptsWithEmptyContent:      'The client script you tried to inject is empty.',
-    clientScriptsWithDuplicatedContent: 'You injected the following client script{suffix} several times:\n {duplicatedScripts}'
+    clientScriptsWithDuplicatedContent: 'You injected the following client script{suffix} several times:\n {duplicatedScripts}',
+    assertedSelectorInstance:           'You are trying to assert the Selector instance.\nIf you want to assert that the selected element exists, consider using yourSelector.exists in the .expect() statement.',
+    assertedClientFunctionInstance:     'You are trying to assert the ClientFunction instance.\nIf you want to assert the returned value, consider calling your ClientFunction (using yourClientFunction.call()) in the .expect() statement.'
```
Comment (VasilyStrelyaev) [inline src/notifications/warning-message.js:37]:
```js
    assertedSelectorInstance:           'You passed a Selector object to \'t.expect()\'.\nIf you want to check that a matched element exists, pass the \'selector.exists\' value instead.',
    assertedClientFunctionInstance:     'You passed a ClientFunction object to \'t.expect()\'.\nIf you want to check the function\'s return value, call the client function (\'clientFunction()\') instead.'

Comment (VasilyStrelyaev) [reply src/notifications/warning-message.js:37]: Checked with Dirk.

Comment (VasilyStrelyaev) [reply src/notifications/warning-message.js:37]: @Ogurecher Please edit the messages in a new PR.

**Pull request merged**